src/ranged.cpp

```diff
@@ -155,7 +155,9 @@ static const std::set<material_id> ferric = { material_iron, material_steel, mat
 static constexpr int AIF_DURATION_LIMIT = 10;
 
 static projectile make_gun_projectile( const item &gun );
-static int time_to_attack( const Character &p, const itype &firing );
+static int NPC_time_to_attack( const Character &p, const itype &firing );
+static int time_to_attack( const Character &p, const item &firing, const item_location &loc );
+static int RAS_time = 0;
```

Comment:
I can see why you would want to do this, but it's not ok, this data needs to be plumbed into the affected methods not stashed like this.

Reply:
OK, I'll rewrite this part :)